### PR TITLE
Updates to userRegistrationDetails API

### PR DESCRIPTION
**Pull request by kvenkit**
This docs PR is tracked via this work item https://dev.azure.com/msft-skilling/Content/_workitems/edit/116956

Updated properties:
1. isSystemPreferredAuthenticationMethodEnabled
2. systemPreferredAuthenticationMethods
3. userPreferredMethodForSecondaryAuthentication

### CODE REVIEW

**Comment by Lauragra** [comment]
@kvenkit , can you please follow the Get started steps to set yourself up to be a contributor? https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211768/Author-content

When you're a contributor, you can follow our content workflow: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211773/Content-workflow

File: changelog/Microsoft.AAD.Reporting.json [hunk 1]
```diff
@@ -1,5 +1,47 @@
 {
   "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "6248c9ab-afef-4d70-81e8-7eb948a3a45b",
+          "ApiChange": "Enumeration",
+          "ChangedApiName": "userDefaultAuthenticationMethod",
+          "ChangeType": "Addition",
+          "Description": "Added the **userDefaultAuthenticationMethod** enumeration type.",
```
Comment (Jarbas-MSFT) [inline changelog/Microsoft.AAD.Reporting.json:10]: I don't see any associated documentation update for this enumeration. Please, update the docs accordingly.

File: api-reference/beta/resources/userregistrationdetails.md [hunk 2]
```diff
@@ -66,9 +66,12 @@ The following is a JSON representation of the resource.
   "isSsprCapable": "Boolean",
   "isSsprEnabled": "Boolean",
   "isSsprRegistered": "Boolean",
+  "isSystemPreferredAuthenticationMethodEnabled": "Boolean",
```
Comment (Jarbas-MSFT) [inline api-reference/beta/resources/userregistrationdetails.md:69]: Please, add a documentation for these three new properties also in the "Properties" table above.

**Comment by Jarbas-MSFT** [comment]
Hi @kvenkit, I started the content review work for your PR. Please take a look at the comments when you will have time. Also, please address the topics mentioned by Laura [here](https://github.com/microsoftgraph/microsoft-graph-docs/pull/21673#issuecomment-1595567659) and Faith in Teams. Thanks!

**Comment by kvenkit** [comment]
#feedback-addressed

File: changelog/Microsoft.AAD.Reporting.json [hunk 3]
```diff
@@ -1,5 +1,47 @@
 {
   "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "6248c9ab-afef-4d70-81e8-7eb948a3a45b",
+          "ApiChange": "Enum type",
+          "ChangedApiName": "userDefaultAuthenticationMethod",
+          "ChangeType": "Addition",
+          "Description": "Added the **userDefaultAuthenticationMethod** enumeration type.",
```
Comment (Jarbas-MSFT) [inline changelog/Microsoft.AAD.Reporting.json:10]: I don't see any associated documentation update for this enumeration. Please, update the docs (https://learn.microsoft.com/en-us/graph/api/resources/enums?view=graph-rest-beta) accordingly.

File: api-reference/beta/resources/userregistrationdetails.md [hunk 4]
```diff
@@ -35,9 +35,12 @@ Inherits from [entity](../resources/entity.md).
 |isSsprCapable|Boolean|Whether the user has registered the required number of authentication methods for self-service password reset and the user is allowed to perform self-service password reset by policy. Supports `$filter` (`eq`).|
 |isSsprEnabled|Boolean|Whether the user is allowed to perform self-service password reset by policy. The user may not necessarily have registered the required number of authentication methods for self-service password reset. Supports `$filter` (`eq`).|
 |isSsprRegistered|Boolean|Whether the user has registered the required number of authentication methods for self-service password reset. The user may not necessarily be allowed to perform self-service password reset by policy. Supports `$filter` (`eq`).|
+|isSystemPreferredAuthenticationMethodEnabled|Boolean|If the system dynamically determines the most secure authentication method, the user is registered with for secondary authentication.|
```
Comment (Jarbas-MSFT) [inline api-reference/beta/resources/userregistrationdetails.md:38]: For clarity, please rephrase this description. Suggested revision:
- "Indicates whether system preferred authentication method is enabled. If the system dynamically determines the most secure authentication method, the user is registered with secondary authentication."

**Comment by kvenkit** [comment]
#feedback-addressed

**Comment by Jarbas-MSFT** [comment]
Hi @Lauragra, hi @FaithOmbongi, this PR is ready for a final review.

**Comment by FaithOmbongi** [comment]
Thanks @Jarbas-MSFT !

File: api-reference/beta/resources/userregistrationdetails.md [hunk 5]
```diff
@@ -35,9 +35,12 @@ Inherits from [entity](../resources/entity.md).
...
 |methodsRegistered|String collection|Collection of authentication methods registered, such as `mobilePhone`, `email`, `fido2`. Supports `$filter` (`any` with `eq`).|
+|systemPreferredAuthenticationMethods|String collection|Collection of authentication methods that the system determined to be the most secure authentication methods among the registered methods for second factor authentication. Possible values are: `push`, `oath`, `voiceMobile`, `voiceAlternateMobile`, `voiceOffice`, `sms`, `none`, `unknownFutureValue`. Supports `$filter` (`any` with `eq`).|
 |userDisplayName|String| The user display name, such as `Adele Vance`. Supports `$filter` (`eq`, `startsWith`) and `$orderBy`.|
+|userPreferredMethodForSecondaryAuthentication|userDefaultAuthenticationMethod|The method the user selected as default for performing multi-factor authentication. Possible values are: `push`, `oath`, `voiceMobile`, `voiceAlternateMobile`, `voiceOffice`, `sms`, `none`, `unknownFutureValue`.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/userregistrationdetails.md:43]: Julia called out the [2nd factor in the auth methods docs](https://learn.microsoft.com/en-us/graph/api/resources/signinpreferences?view=graph-rest-beta), please confirm this change is accurate
```suggestion
|userPreferredMethodForSecondaryAuthentication|userDefaultAuthenticationMethod|The method the user selected as the default second-factor for performing multi-factor authentication. Possible values are: `push`, `oath`, `voiceMobile`, `voiceAlternateMobile`, `voiceOffice`, `sms`, `none`, `unknownFutureValue`.|
```

File: api-reference/beta/resources/enums.md [hunk 6]
```diff
@@ -3963,3 +3963,16 @@ Possible values for user account types (group membership), per Windows definitio
...
+| Member                     |
+| :------------------------- |
+| push                       |
+| oath                       |
+| voiceMobile                |
+| voiceAlternateMobile       |
+| voiceOffice                |
+| sms                        |
+| none                       |
+| unknownFutureValue         |
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/enums.md:3978]: Duplicate entry. The enum was documented when we published the core auth methods docs.
```suggestion
```

File: changelog/Microsoft.AAD.Reporting.json [hunk 7]
```diff
@@ -1,5 +1,47 @@
 {
   "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "6248c9ab-afef-4d70-81e8-7eb948a3a45b",
+          "ApiChange": "Enum type",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:7]: This was actually added in the core auth methods docs but I guess it doesn't hurt.

**Review by FaithOmbongi — APPROVED** [review]
Hi @kvenkit -

I've left two suggestions inline with the related core auth methods APIs. Please address those before setting this PR to ready to merge.

Approved with comments.

**Pull request merged**